### ISSUE TITLE
Generated declaration file contains errors when export class as a type

### ISSUE THREAD

**Issue by foisonocean** [wessberg/rollup-plugin-ts#68]
source code:

```typescript
// src/a.ts
export class A {}


// src/index.ts
import { A as AClass } from './a';

type A = AClass;

export default class B {}

export { A }
```

generated declaration file:

```typescript
declare class A {
}
declare const AClass: typeof A;
type A$0 = AClass;
declare class B {
}
export { B as default, A$0 as A };
```
 which contains an error: 'AClass' refers to a value, but is being used as a type here.

**Comment by wessberg** [comment]
Hey there. Based on the example you provided, the error was already present in `src/index.ts` where you declare the TypeAliasDeclaration:

```typescript
type A = AClass;
```

Which is an error. That error will naturally also be present in the bundles output. Is there another example that will lead to the bug where the error isn't already present in the source code?

**Comment by foisonocean** [comment]
@wessberg 
Thanks for your reply, but actually `type A = AClass` is valid in TypeScript, I use `tsc` to check the source code and there is no errors reported from `tsc`.

Class in Typescript is a little bit different compare to other things like functions or variables. Functions and variables are values in TypeScript, if you want to use the type of them, you need to use `type A = typeof myFn`. But class is value and type at the same, for example `let a = new A()`, `a`'s type is `A`.

What I do in the previous source code, is just prevent user to access the value of the class `A`. When user want import `A` in `src/index.ts`, he/she can only import the type of it.

**Comment by wessberg** [comment]
@foisonocean, I stand corrected, I was under the assumption that `AClass` was pointing to the identifier of a VariableDeclaration such as in the declaration output, but I now see under further inspection of your original example that it is indeed pointing directly to a ClassDeclaration. It is broken in the declaration output because `A` is aliased to a VariableDeclaration, `AClass`, and that's when it loses the special trait of both being a value and a type.

It's gonna be a tricky fix, but I'll look into it.

**Comment by weyert** [comment]
Hmm, I think I have a similar issue where it doesn't export from component props type. I will see if I can reproduce it or at least make a ticket out of it.

**Comment by wessberg** [comment]
Given that [Type Aliases cannot be instantiated](https://github.com/microsoft/TypeScript/issues/2559) and that [type aliases don't work properly with classes](https://github.com/Microsoft/TypeScript/issues/2552), I think the best way forward is to implement  [the example suggested by @mhegazy](https://github.com/Microsoft/TypeScript/issues/2552#issuecomment-87879737) in which an `ImportEqualsDeclaration` is being used.

That would mean that the example above would turn into the following:

```typescript
declare class A {
}
declare module AModule {
    export { A };
}
import AClass = AModule.A;
type A$0 = AClass;
export { A$0 as A };
```

It would be very nice if the `ModuleDeclaration` re-exporting the class could be avoided, but it seems that it's not possible. It's not that pretty, but it seems to be the only way to correctly preserve the class. An alternative I considered was:

```typescript
declare class A {
}
declare class AClass extends A {}
type A$0 = AClass;
export { A$0 as A };
```

And while it is more compact, it is not semantically equivalent, so I think I will proceed with the first code example as the fix for this one.

**Comment by wessberg** [comment]
A fix has been released in [v1.2.7](https://github.com/wessberg/rollup-plugin-ts/releases/tag/v1.2.7)
